.travis.yml

```diff
@@ -46,27 +49,32 @@ jobs:
           && mvn clean install -q -f lib -DskipTests -Dmaven.javadoc.skip=true
           && mvn clean install -q -f rest -DskipTests -Dmaven.javadoc.skip=true
-    - stage: "Release / Deploy"
-      script: 
-        - ci/change_version.sh -i -m "lib/ rest/" -swarm lib/taskana-cdi/src/test/java/pro/taskana/TaskanaProducersTest.java
-          && ci/copy-rest-spring.sh $TRAVIS_TAG
-          && ci/commitPoms.sh lib/taskana-cdi/src/test/java/pro/taskana/TaskanaProducersTest.java
-      deploy: 
-        provider: cloudfoundry 
-        username: tobias.schaefer@novatec-gmbh.de
-        password:
-          secure: dQh3e8Zhi2ucd5NcxHZ337Ku2SFrt0qA8lghtExQU5bcdFziqYKJsVFdD19hNQE6DOZT8FV9OHTB3odZSvV7aa38sq4KDwzJVVitIyU3eNZP0S5AqAv0eKitqQhjY7rnjBIkgMQ2rGTZlAL9g/CQWPlB4rnUPLXv/7+pZP29M3OPK0hyIPw9I0FfU02cM+R2z8SYPRr4MYYbdsTdJeJz+1sKWwQiD4bDE/Kmcs7BALTZ3/HFmu2Kjxk7soERSpXlMvKDkHHJCJMSjQI6zT5Ja6yg4SgOgccLZKuaPwF+8a7RKPOtW+H74zzO/IE9FGimlRKFGHhoFmQh0oI1sMkhRyIH0n/rkCVzI6B8XjDBEdaRjfHZbAkrjapuvhLwSxyklut3CknbSxfgAo8tZefsGVETD1rYkfLA207bkxt6Wk+X1d2pCRFkUoeTjcB+IkqvftTJvTX5vyt927w0pyx/xg9gxBLA3bqzhRL7rwgXgqgrxNUDFT3nxExecPpdEinyVf+jQxmrb1uw1WEFM+dLjA3tByxMEy7PfnfqTvUs8dvs4em1Vln+N5K1+OqU93ze3FmPMmqY3/4r63qdVACtCqHlfGC3HQPUi6fWmBzVNhZ9rF6FJvkECTkvLklnOB9MIoneGk2v46slveUsxKSz985A2aMkYNO+p1VozS8rAfg=
-        api: https://api.ng.bluemix.net 
-        organization: '"NovaTec Consulting GmbH"' 
-        space: Taskana
-        on:
-          all_branches: true
-      if: repo = env(DEPLOY_REPO) AND tag =~ ^v[0-9]+\.[0-9]+\.[0-9]+$
-    - stage: "Release / Deploy"
+    - stage: Release
       script:
         - ci/release.sh lib/taskana-core $TRAVIS_TAG
           && ci/release.sh lib/taskana-spring $TRAVIS_TAG
           && ci/release.sh lib/taskana-cdi $TRAVIS_TAG
           && ci/release.sh web/ $TRAVIS_TAG
           && ci/release.sh rest/taskana-rest-spring $TRAVIS_TAG
       if: repo = env(DEPLOY_REPO) AND (tag =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ OR branch = master) AND type != pull_request
+    - stage: AfterRelease
+      script: 
+        - ci/change_version.sh -i -m "lib/ rest/" -swarm lib/taskana-cdi/src/test/java/pro/taskana/TaskanaProducersTest.java
+          && ci/commitPoms.sh lib/taskana-cdi/src/test/java/pro/taskana/TaskanaProducersTest.java
+      if: repo = env(DEPLOY_REPO) AND tag =~ ^v[0-9]+\.[0-9]+\.[0-9]+$
+    - stage: Deploy
+      script: 
+        - if [[ $FORCE_DEPLOY_TAG =~ ^v[0-9]+\.[0-9]+\.[0-9] ]];
+            then ci/copy-rest-spring.sh $FORCE_DEPLOY_TAG;
+            else ci/copy-rest-spring.sh $TRAVIS_TAG;
+          fi  
+      deploy:
+        provider: cloudfoundry
+        username: $BLUEMIX_ACCOUNT
+        password: $BLUEMIX_ACCOUNT_PASSWORD
+        api: https://api.ng.bluemix.net
+        organization: "NovaTec Consulting GmbH"
+        space: Taskana
+        on:
+          all_branches: true
+      if: (repo = env(DEPLOY_REPO) AND tag =~ ^v[0-9]+\.[0-9]+\.[0-9]+$) OR (env(FORCE_DEPLOY_TAG) =~ ^v[0-9]+\.[0-9]+\.[0-9])
```

Comment:
It is not possible to have a tag during a PR. That's why I didn't check for it here.
Is it possible to custom trigger a PR build? If yes, then please catch that - We never ever want to deploy a PR build :)

Reply:
That's depend on the purpose of the production web application if  it is really a production application or it is just a demo application, If it is a demo I don't see why we don't want to deploy PR, if is for production  purpose we should only deply master.

Reply:
@holgerhagen I think Miguel and I are not in a position to decide this. What's your call?

Reply:
A PR needs to be integrated before it makes sense to deploy it, or am I getting something wrong here? We would only deploy the master branch which would explude the PRs, wouldn't it?

Reply:
That's all I wanted to hear. - Let's make the restriction that the branch hast to be master and not a pr